### PR TITLE
Chore: Migrate i18n to Typescript 

### PR DESCRIPTION
**Pull request by reinaldonetof**
## How to test or reproduce

- Check if the translations are work properly
- Go to Settings -> Language -> change between the languages

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/List/ListItem.tsx [hunk 1]
```diff
@@ -91,7 +91,7 @@ const Content = React.memo(
 				<View style={styles.textContainer}>
 					<View style={styles.textAlertContainer}>
 						<Text style={[styles.title, { color: color || themes[theme].titleText }]} numberOfLines={1}>
-							{translateTitle ? I18n.t(title) : title}
+							{translateTitle ? I18n.t(title || '') : title}
```
Comment (diegolmello) [inline app/containers/List/ListItem.tsx:94]: I've seen this on some other places on this PR.
If the code reaches `''`, it's actually going to show to the end user an awkward message [like this](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3960).
A better way to fix this warning is:

```suggestion
							{translateTitle && title ? I18n.t(title) : title}
```

Please review your PR and change it on other places as well.

File: app/containers/message/utils.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { TMessageModel } from '../../definitions/IMessage';
 import I18n from '../../i18n';
 import { DISCUSSION } from './constants';
 
-export const formatMessageCount = (count?: number, type?: string): string => {
+export const formatMessageCount = (count?: number, type?: string) => {
```
Comment (diegolmello) [inline app/containers/message/utils.ts:5]: I'm sure this is a better approach
```suggestion
export const formatMessageCount = (count?: number, type?: string): string | null => {
```

File: app/i18n/index.ts [hunk 3]
```diff
@@ -104,6 +114,7 @@ export const setLanguage = l => {
 	i18n.translations = { ...i18n.translations, [locale]: translations[locale]?.() };
 	I18nManager.forceRTL(isRTL(locale));
 	I18nManager.swapLeftAndRightInRTL(isRTL(locale));
+	// @ts-ignore
```
Comment (diegolmello) [inline app/i18n/index.ts:117]:
```suggestion
	// @ts-ignore TODO: we need to review this logic
```

File: app/views/DiscussionsView/index.tsx [hunk 4]
```diff
@@ -130,10 +130,6 @@ const DiscussionsView = ({ navigation, route }: IDiscussionsViewProps): JSX.Elem
 			),
 			headerTitleAlign: 'center',
 			headerTitle: I18n.t('Discussions'),
-			headerTitleContainerStyle: {
```
Comment (diegolmello) [inline app/views/DiscussionsView/index.tsx:133]: Same as the image style.
Please stop removing stuff like this.

**Review by diegolmello — CHANGES_REQUESTED** [review]

**Comment by GleidsonDaniel** [comment]
Adding one more comment to what @diegolmello  said.
If the migration is about i18n, I don't think it's ideal to change things that are unrelated to i18n, even if they look wrong. Better would be to create a task to change this in the future.

PR consistency is very important for tracking.

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]
The i18 part is ok, but as we said, it will remove the part that changes other things and open another PR with these changes.

**Comment by reinaldonetof** [comment]
> The i18 part is ok, but as we said, it will remove the part that changes other things and open another PR with these changes.

Hey, @GleidsonDaniel I created a new function called `isTranslated` to be a function to test if the string to translate exists. Also, I could merge the develop here and deleted the files from

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**